samples/bluetooth/unicast_audio_client/src/main.c

```diff
@@ -0,0 +1,531 @@
+NET_BUF_POOL_FIXED_DEFINE(tx_pool, 1, CONFIG_BT_ISO_TX_MTU + BT_ISO_CHAN_SEND_RESERVE, NULL);
+
+#define LC3_PRESET(_name, _codec, _qos) \
+	{ \
+		.name = _name, \
+		.codec = _codec, \
+		.qos = _qos, \
+	}
+
+struct lc3_preset {
+	const char *name;
+	struct bt_codec codec;
+	struct bt_codec_qos qos;
+};
+
+
+/* Mandatory support preset by both client and server */
+struct lc3_preset preset_16_2_1 =
+	LC3_PRESET("16_2_1",
+		   BT_CODEC_LC3_CONFIG_16_2,
+		   BT_CODEC_LC3_QOS_10_INOUT_UNFRAMED(40u, 2u, 10u, 40000u));
+
+
+
+void print_hex(const uint8_t *ptr, size_t len)
+{
+	while (len-- != 0) {
+		printk("%02x", *ptr++);
+	}
+}
+
+static void print_codec(const struct bt_codec *codec)
+{
+	printk("codec 0x%02x cid 0x%04x vid 0x%04x count %u\n",
+	       codec->id, codec->cid, codec->vid, codec->data_count);
+
+	for (size_t i = 0; i < codec->data_count; i++) {
+		printk("data #%zu: type 0x%02x len %u\n",
+		       i, codec->data[i].data.type,
+		       codec->data[i].data.data_len);
+		print_hex(codec->data[i].data.data,
+			  codec->data[i].data.data_len -
+				sizeof(codec->data[i].data.type));
+		printk("\n");
+	}
+
+	for (size_t i = 0; i < codec->meta_count; i++) {
+		printk("meta #%zu: type 0x%02x len %u\n",
+		       i, codec->meta[i].data.type,
+		       codec->meta[i].data.data_len);
+		print_hex(codec->meta[i].data.data,
+			  codec->meta[i].data.data_len -
+				sizeof(codec->meta[i].data.type));
+		printk("\n");
+	}
+}
+
+static void print_qos(struct bt_codec_qos *qos)
+{
+	printk("QoS: dir 0x%02x interval %u framing 0x%02x phy 0x%02x sdu %u "
+	       "rtn %u latency %u pd %u\n",
+	       qos->dir, qos->interval, qos->framing, qos->phy, qos->sdu,
+	       qos->rtn, qos->latency, qos->pd);
+}
+
+/**
+ * @brief Send audio data on timeout
+ *
+ * This will send an increasing amount of audio data, starting from 1 octet.
+ * The data is just mock data, and does actually represent any audio.
+ *
+ * First iteration : 0x00
+ * Second iteration: 0x00 0x01
+ * Third iteration : 0x00 0x01 0x02
+ *
+ * And so on, until it wraps around the configured MTU (CONFIG_BT_ISO_TX_MTU)
+ *
+ * @param work Pointer to the work structure
+ */
+static void audio_timer_timeout(struct k_work *work)
+{
+	int ret;
+	static uint8_t buf_data[CONFIG_BT_ISO_TX_MTU];
+	static bool data_initialized;
+	struct net_buf *buf;
+	static size_t len_to_send = 1;
+
+	if (!data_initialized) {
+		/* TODO: Actually encode some audio data */
+		for (int i = 0; i < ARRAY_SIZE(buf_data); i++) {
+			buf_data[i] = (uint8_t)i;
+		}
+
+		data_initialized = true;
+	}
+
+	buf = net_buf_alloc(&tx_pool, K_FOREVER);
+	net_buf_reserve(buf, BT_ISO_CHAN_SEND_RESERVE);
+
+	net_buf_add_mem(buf, buf_data, len_to_send);
+
+	ret = bt_audio_chan_send(&audio_channel, buf);
+	if (ret < 0) {
+		printk("Failed to send audio data (%d)\n", ret);
+		net_buf_unref(buf);
+	} else {
+		printk("Sending mock data with len %zu\n", len_to_send);
+	}
+
+	k_work_schedule(&audio_send_work, K_MSEC(1000));
```

Comment:
Do we have `chan->send_end_cb` now? It’s more appropriate to put it here. Of course, our current interval is large enough, and there’s nothing wrong with it.

Reply:
We don't have a send callback for audio just yet, but since we added that to ISO we can definitely add it here - Good idea! 

For this sample, with just sending data in an interval (which is what most audio application would likely do), I think just using a timer is still overall better.